src/settingsMigrator.ts

```diff
@@ -1,7 +1,12 @@
-import {TodoistSettings} from "./DefaultSettings";
+import {TodoistSettings, keywordTodoistQuery} from "./DefaultSettings";
 
 export function migrateSettings(settings: any) : TodoistSettings {
 	let newSettings : any = settings;
+
+	if (getSettingsVersion(newSettings) == 1) {
```

Comment:
pretty sure you'll want this new `if` block to go _after_ line 10, in case someone is upgrading from v0 to v2 - that way they can get both migrations applied sequentially.

In the current code, if someone goes from app v0 to app v2, they'd end up with settings migration from v0 to v1

there's definitely a more algorithmic way to write this function (e.g. a for-loop to automatically run from current version to max available version), up to you if that's worth doing (no pressure from my end)